pymilvus/client/async_grpc_handler.py

```diff
@@ -117,6 +123,15 @@ def __exit__(self: object, exc_type: object, exc_val: object, exc_tb: object):
     async def close(self):
         await self._async_channel.close()
         self._async_channel = None
+        # Automatically remove connection from pool when channel is closed
+        if self._connection_alias:
+            try:
+                connections._alias_handlers.pop(self._connection_alias, None)
+                connections._alias_config.pop(self._connection_alias, None)
+            except Exception as e:
+                logger.debug(
+                    f"Error removing connection {self._connection_alias} during cleanup: {e}"
+                )
```

Comment:
Why we need call back? I think AsyncMilvusClient close() already does the same thing. The call stack should be  AsyncClient.close -> connections.async_remove_connections -> asyncgrpchander.close() why reverse here?